New package: Prometheus.WindowsExporter version 0.30.0

Dragon1573 opened the pull request:
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #213642

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.9 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.9.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

## Manual validation

- For `[x64, user, portable]`, see <https://github.com/microsoft/winget-pkgs/issues/213642#issuecomment-2599800226>
- For `[x64, machine, wix]`, see <https://github.com/microsoft/winget-pkgs/issues/213642#issuecomment-2599803525>

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/213700)

Dragon1573 committed:
New package: Prometheus.WindowsExporter version 0.30.0

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

jkroepke commented on manifests/p/Prometheus/WindowsExporter/0.30.0/Prometheus.WindowsExporter.installer.yaml:
```diff
@@ -0,0 +1,48 @@
+    ProductCode: '{A7B06148-F35F-485E-A77E-5787916498D8}'
+    UpgradeCode: '{66A6EB5B-1FC2-4B14-A362-5CEEC6413308}'
+  InstallationMetadata:
+    DefaultInstallLocation: '%ProgramFiles%\windows_exporter\textfile_inputs'
```
```suggestion
    DefaultInstallLocation: '%ProgramFiles%\windows_exporter\'
```

Not sure what this exactly means, but the binary is placed here. and `textfile_inputs` is just an directory for system admins.

jkroepke commented on manifests/p/Prometheus/WindowsExporter/0.30.0/Prometheus.WindowsExporter.installer.yaml:
```diff
@@ -0,0 +1,48 @@
+    ProductCode: '{9CC45469-E84A-4532-8CC7-E82DBB01B16B}'
+    UpgradeCode: '{66A6EB5B-1FC2-4B14-A362-5CEEC6413308}'
+  InstallationMetadata:
+    DefaultInstallLocation: '%ProgramFiles%\windows_exporter\textfile_inputs'
```
Same question here.

Dragon1573 replied:
```suggestion
    DefaultInstallLocation: '%ProgramFiles%\windows_exporter\'
```

jkroepke commented on manifests/p/Prometheus/WindowsExporter/0.30.0/Prometheus.WindowsExporter.locale.en-US.yaml:
```diff
@@ -0,0 +1,27 @@
+PackageUrl: https://github.com/prometheus-community/windows_exporter
+License: MIT
+LicenseUrl: https://github.com/prometheus-community/windows_exporter/blob/master/LICENSE
+Copyright: Copyright (c) 2016 Martin Lindhe
```
I recently update the upgrade note, the current holder are `Copyright (c) 2021 The Prometheus Authors`. It's already part of all source code headers, but it seems to be forgotten on the LICENSE file.

https://github.com/prometheus-community/windows_exporter/blob/faa98d27087af1035d993074b444fec34e9d957b/LICENSE#L4

Dragon1573 replied:
```suggestion
Copyright: Copyright (c) 2021 The Prometheus Authors
```

jkroepke commented on manifests/p/Prometheus/WindowsExporter/0.30.0/Prometheus.WindowsExporter.locale.en-US.yaml:
```diff
@@ -0,0 +1,27 @@
+Description: A Prometheus exporter for Windows machines.
+Moniker: windows_exporter
+Tags:
+- hacktoberfest
```
The tag `hacktoberfest` is reference in the source code [context](https://hacktoberfest.com/), but does not make sense having this tag on a package manager.

Dragon1573 replied:
```suggestion
```

jkroepke commented on manifests/p/Prometheus/WindowsExporter/0.30.0/Prometheus.WindowsExporter.installer.yaml:
```diff
@@ -0,0 +1,48 @@
+  InstallerUrl: https://github.com/prometheus-community/windows_exporter/releases/download/v0.30.0/windows_exporter-0.30.0-arm64.msi
+  InstallerSha256: 3C2BB102FDA2206E022F6A522BBB1F6E07659B1125D96BB5C85E15103D2113E0
+  ProductCode: '{9CC45469-E84A-4532-8CC7-E82DBB01B16B}'
+  AppsAndFeaturesEntries:
```
With features, does it referenced to msi features? If yes, there is a msi feature named [`FirewallException`](https://github.com/prometheus-community/windows_exporter/blob/faa98d27087af1035d993074b444fec34e9d957b/installer/main.wxs#L131) which allows admins to add an firewall exception for the windows_exporter. Thats useful, since the http endpoints is called from external systems.

Dragon1573 replied:
No, it's not related to some "MSI features". It's for ARP ("Add & Remove Programs") entries information.

Following properties are only acceptable in this section. See [the doc](https://github.com/microsoft/winget-pkgs/blob/master/doc/manifest/schema/1.9.0/installer.md).

```yaml
AppsAndFeaturesEntries:        # Optional entries from the Add and Remove Programs (ARP) table
- DisplayName:                 # Optional program name shown in the ARP entry
  DisplayVersion:              # Optional version displayed in the ARP entry
  Publisher:                   # Optional publisher displayed in the ARP entry
  ProductCode:                 # Optional product code of the installer
  UpgradeCode:                 # Optional upgrade code of the installer
  InstallerType:               # Optional installer type
```

Dragon1573 committed:
Apply suggestions from code review

- Change the "Default Install Location" property
- Correction of "Copyright" property
- Remove "non-sense" value from "Tags" property

Co-authored-by: Jan-Otto Kröpke <github@jkroepke.de>

Dragon1573 committed:
Apply suggestions from code review

- Rename portable binary, as instructed in #213642

Co-authored-by: Jan-Otto Kröpke <github@jkroepke.de>

stephengillie commented:
Automatic Validation ended with:
>  Executable C:\Program Files\windows_exporter\windows_exporter.exe returned exit code: 1

(Automated response - build 949.)

stephengillie commented:
I'm pretty sure that the https://github.com/microsoft/winget-pkgs/labels/Policy-Test-2.7 label is accidental, from removing the space between "Windows" and "Exporter". A well-known answers site has the same issue.

jkroepke commented:
As someone, who would like to understand the eco system. How I should interpret this? From my point of view, the YAML file looks fine.

Dragon1573 commented:
✅ Manual validate again.

ℹ️ Note: I have trouble connect to the origin download URL, so I manually download the file separately and save it on my local "cache" server.

```text
--> Installing WinGet
--> Disabling safety warning when running installers
Tip: you can type 'Update-EnvironmentVariables' to update your environment variables, such as after installing a new software.

--> Configuring Winget
Enabled admin setting 'LocalManifestFiles'.
Enabled admin setting 'LocalArchiveMalwareScanOverride'.

--> Installing the Manifest 0.30.0

Found windows_exporter [Prometheus.WindowsExporter] Version 0.30.0
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading http://alist.dragon1573.local/d/windows_exporter-0.30.0-amd64.msi
  ██████████████████████████████  11.4 MB / 11.4 MB
Successfully verified installer hash
Starting package install...
Successfully installed

--> Refreshing environment variables

--> Comparing ARP Entries

DisplayName      DisplayVersion Publisher            ProductCode                            Scope
-----------      -------------- ---------            -----------                            -----
windows_exporter 0.30.0         prometheus-community {A7B06148-F35F-485E-A77E-5787916498D8} Machine

PS C:\Users\WDAGUtilityAccount\Desktop\winget-pkgs> & 'C:\Program Files\windows_exporter\windows_exporter.exe'
time=2025-01-23T16:21:19.959Z level=WARN source=cs.go:75 msg="The cs collector is deprecated and will be removed in a future release. Logical processors has been moved to cpu_info collector. Physical memory has been moved to memory collector. Hostname has been moved to os collector."
time=2025-01-23T16:21:19.959Z level=WARN source=os.go:108 msg="The os collect holds a number of deprecated metrics and will be removed mid 2025. See https://github.com/prometheus-community/windows_exporter/pull/1596 for more information." collector=os
time=2025-01-23T16:21:20.159Z level=INFO source=main.go:311 msg="Running as C9AAF6AC-D13D-4\\WDAGUtilityAccount"
time=2025-01-23T16:21:20.159Z level=INFO source=main.go:228 msg="Enabled collectors: cpu, cs, memory, logical_disk, physical_disk, net, os, service, system"
time=2025-01-23T16:21:20.160Z level=INFO source=main.go:246 msg="starting windows_exporter in 214.6886ms" version=0.30.0 branch=HEAD revision=86e6d12518d2fa7f8f2f632634ade3aaa34fba22 goversion=go1.23.4 builddate=20250114-22:51:51 maxprocs=28
time=2025-01-23T16:21:20.160Z level=ERROR source=main.go:283 msg="Failed to start windows_exporter" err="listen tcp :9182: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted."
```

It seems the server will be automatically launched after installation.

```text
PS C:\Users\WDAGUtilityAccount\Desktop\winget-pkgs> Invoke-RestMethod -Uri http://localhost:9182/health

status
------
ok


PS C:\Users\WDAGUtilityAccount\Desktop\winget-pkgs> Invoke-RestMethod -Uri http://localhost:9182/metrics
# HELP go_build_info Build information about the main Go module.
# TYPE go_build_info gauge
go_build_info{checksum="",path="github.com/prometheus-community/windows_exporter",version="(devel)"} 1
# HELP go_gc_duration_seconds A summary of the wall-time pause (stop-the-world) duration in garbage collection cycles.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 0
go_gc_duration_seconds{quantile="0.25"} 0
go_gc_duration_seconds{quantile="0.5"} 0
go_gc_duration_seconds{quantile="0.75"} 5.95e-05
go_gc_duration_seconds{quantile="1"} 5.95e-05
go_gc_duration_seconds_sum 5.95e-05
go_gc_duration_seconds_count 4
...... (Other stuff ignored)
```

jkroepke commented:
> It seems the server will be automatically launched after installation.

the msi installs a service which starts the server. this is correct.

stephengillie commented:
Verify:
![image](https://github.com/user-attachments/assets/dbc9e9a3-438e-4f67-bd45-1ee7498bbb26)

stephengillie commented:
> As someone, who would like to understand the eco system. How I should interpret this? From my point of view, the YAML file looks fine.

I'm not exactly sure what happened here, but I'm guessing a bad value got into the manual validation manifest.

jkroepke commented:
Thanks. One question about new versions: is the process automated? Do I have to create an issue each time?
